addons/test_mail/tests/test_mail_mail.py

```diff
@@ -824,6 +831,97 @@ def test_mail_mail_values_unicode(self):
+        - with attachment owned by a business record or not (attachments not owned by a
+        business record are never turned into links because they are deleted after the
+        mail is sent)
+        """
+        def count_attachments(message):
+            return (0 if isinstance(message, str)
+                    else sum(count_attachments(part) for part in message.get_payload()) if message.is_multipart()
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+        # Define some constant to ease the understanding of the test
+        using_specific_mail_server = self.mail_server_domain_2
+        using_default_mail_server = None
+        cst_max_size_always_exceed = 0.1
+        cst_max_size_never_exceed = 10
+
+        # cst_max... are used to define max_email_size on the server that is not used with value that would
+        # lead to an incorrect expected_is_links if they were used. We do this to check that the right max
+        # email size is used.
+        for (n_attachment, mail_server, default_email_size, max_email_size, business_attachment), expected_is_links in (
+                # 1 attachment which doesn't exceed max size
+                ((1, using_default_mail_server, 0.3, cst_max_size_always_exceed, True), False),
+                # 3 attachment: exceed max size
+                ((3, using_default_mail_server, 0.3, cst_max_size_never_exceed, True), True),
+                # 1 attachment: exceed max size
+                ((1, using_default_mail_server, 0.1, cst_max_size_never_exceed, True), True),
+                # Same as above with a specific server. Note that the default and server max_email size are reversed.
+                ((1, using_specific_mail_server, cst_max_size_always_exceed, 0.3, True), False),
+                ((3, using_specific_mail_server, cst_max_size_never_exceed, 0.3, True), True),
+                ((1, using_specific_mail_server, cst_max_size_never_exceed, 0.1, True), True),
+                # Attachments not linked to a business record are never turned to link
+                ((3, using_default_mail_server, 0.3, cst_max_size_never_exceed, False), False),
+                ((1, using_specific_mail_server, cst_max_size_never_exceed, 0.1, False), False),
+        ):
+            self.env['ir.config_parameter'].sudo().set_param('mail.outgoing_email_size', default_email_size)
+            if mail_server:
+                mail_server.max_email_size = max_email_size
+            attachments = self.env['ir.attachment'].sudo().create([{
+                'name': f'attachment{idx_attachment}',
+                'res_name': 'test',
+                'res_model': self.test_record._name if business_attachment else False,
+                'res_id': self.test_record.id if business_attachment else 0,
```

Comment:
Won't other attachments be linked to the mail.mail or mail.message (e.g. mass mailing case) ? Shouldn't we mimic that behavior ?